src/SocialAuthUserManager.php

```diff
@@ -139,36 +142,71 @@ public function setSessionKeysToNullify(array $session_keys) {
   /**
    * Creates and/or authenticates an user.
    *
-   * @param string $email
-   *   The user's email address.
    * @param string $name
-   *   The user's name.
-   * @param string $id
-   *   The user's id in provider.
+   * The user's name.
+   * @param string $email
+   * The user's email address.
+   * @param string $type
+   * The social implementer identifier.
+   * @param string $social_media_id
+   * The unique id returned by the user.
    * @param string|bool $picture_url
-   *   The user's picture.
+   * The user's picture.
    *
    * @return \Symfony\Component\HttpFoundation\RedirectResponse
-   *   A redirect response.
+   * A redirect response.
    */
-  public function authenticateUser($email, $name, $id = NULL, $picture_url = FALSE) {
-    // Tries to load the user by their email.
-    $drupal_user = $this->loadUserByProperty('mail', $email);
-    // If user email has already an account in the site.
-    if ($drupal_user) {
-      // Authenticates and redirect existing user.
-      return $this->authenticateExistingUser($drupal_user);
+  public function authenticateUser($name, $email, $type, $social_media_id, $picture_url = FALSE) {
+
+    // Get ID of logged in user.
+    $uid = \Drupal::currentUser()->id();
+
+    if($uid){
+      return $this->addUserRecord($uid(), $type, $social_media_id);
```

Comment:
I think it is better to check if the user has the role 'authenticated' with the method hasPermission() of User class